test/discrepancies/Nonce.js

```diff
@@ -299,12 +342,586 @@ describe('@discrepancies - Nonce Test Suite', async () => {
+    const fundTx2 = await signers[0].sendTransaction({
+      to: wallet.address,
+      value: Utils.tinybarToWeibarCoef // 1 tinybar
+    });
+    await fundTx2.wait();
+
+    //verify nonce is 0
+    const snAfterNewCreate = await getServicesNonce(wallet.address);
+    const mnAfterNewCreate = await getMirrorNodeNonce(wallet.address);
+
+    expectNonIncrementedNonce(snAfterNewCreate, mnAfterNewCreate, 0, 0);
+  });
+
+  //NONCE-022
+  //should fix the get contract nonce function - waiting for SDKs
+  xit('should update all nonces when chained deploys of contracts', async function () {
+    //deploys contract A which deploys contract B which deploys contract C
+    //nonces of signer, contract A and contract B should increment
+    //nonce of contract C should be 0
+    const snBefore = await getServicesNonce(signers[0].address);
+    const mnBefore = await getMirrorNodeNonce(signers[0].address);
+
+    //deploy contract A which will deploy Contract B which will deploy Contract C
+    const chainedContractsFactory = await ethers.getContractFactory(Constants.Contract.ChainedContracts);
+    const chainedContracts = await chainedContractsFactory.deploy({ gasLimit: 5_000_000 });
+    const receipt = await chainedContracts.deploymentTransaction().wait();
+
+    console.log(receipt.logs);
+
+    //get the 3 contracts addresses
+    const innerContract = receipt.logs[0].address;
+    const intermediaryContract = receipt.logs[1].address;
+    const outerContract = receipt.logs[2].address;
+
+    //get the 3 contracts nonces
+    const servicesInnerContractNonce = await Utils.getContractInfo(innerContract, sdkClient);
+    const mirrorNodeInnerContractNonce = await getMirrorNodeNonce(innerContract);
+
+    const servicesIntermediaryContractContractNonce = await getServicesContractNonce(intermediaryContract.target);
+    const mirrorNodeIntermediaryContractNonce = await getMirrorNodeNonce(intermediaryContract.target);
+
+    const servicesOuterContractNonce = await getServicesContractNonce(outerContract.target);
+    const mirrorNodeOuterContractNonce = await getMirrorNodeNonce(outerContract.target);
+
+    //verify signer nonces have updated correctly
+    const snAfter = await getServicesNonce(signers[0].address);
+    const mnAfter = await getMirrorNodeNonce(signers[0].address);
+    expectIncrementedNonce(snBefore, mnBefore, snAfter, mnAfter);
+
+    //verify contract nonces have updated 
+    expectIncrementedNonce(0, 0, servicesIntermediaryContractContractNonce, mirrorNodeIntermediaryContractNonce);
+    expectIncrementedNonce(0, 0, servicesOuterContractNonce, mirrorNodeOuterContractNonce);
+
+    //verify nonce of the inner most contract is 0
+    expectNonIncrementedNonce(0, 0, servicesInnerContractNonce, mirrorNodeInnerContractNonce);
+  });
+
+  //NONCE-023
+  //need to update the get contract nonce - waiting for SDK
+  xit('should update contract nonce for each deployed contract', async function () {
```

Comment:
Just a lil reminder to enable this later (and other tests waiting for the SDK changes)